### PR TITLE
WIP: Add rupture storage

### PR DESCRIPTION
**Pull request by cossatot**
Option to allow reading/writing ruptures to/from disk.  Work primarily by @kejohnso

### CODE REVIEW

File: openquake/hme/core/core.py [hunk 1]
```diff
@@ -223,7 +236,15 @@ def load_inputs(cfg: dict) -> Tuple[GeoDataFrame]:
         Configuration for the evaluations, such as that parsed from the YAML
         config file.
     """
-    rupture_gdf = load_ruptures_from_ssm(cfg)
+
+    rupture_file = cfg["input"]["ssm"]["rupture_file"]
```
Comment (cossatot) [inline openquake/hme/core/core.py:240]: This fails with a KeyError if it isn't there.  There needs to be a try/except block.

**Comment by micheles** [comment]
Serialization of ruptures is non-trivial and it should be done at the engine level, not here. Let us discuss what you want to do, because this is relevant also for other projects, like this one: https://github.com/gem/oq-engine/issues/5603

**Comment by cossatot** [comment]
Thanks @micheles,

We use a [`SimpleRupture`](https://github.com/GEMScienceTools/hamlet/blob/8bca21f5cbf4c7daeedfa087a5154d6994ce06a6/openquake/hme/utils/simple_rupture.py#L38) class in Hamlet which is point-based, just storing the hypocenter, kinematic information, magnitude and occurrence rate.  This minimizes RAM usage which is important when all the ruptures are kept in memory, especially since we use personal computers and not a cluster for Hamlet so far.

**Comment by micheles** [comment]
I am saying that after the governing board we should schedule a meeting with the involved people, because you risk duplicating features that are already in the engine and work already in the general case.

**Pull request merged**